docs/rules/numeric-separators-style.md

```diff
@@ -0,0 +1,86 @@
+# Enforce the style of numeric separators by correctly grouping digits
+
+Enforces a convention of grouping digits using [numeric separators](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#Numeric_separators).
+
+This rule is fixable.
+
+## Fail
+
+```js
+const foo = 1_23_4444;
+const foo = 1_234.56789;
+const foo = 0xAB_C_D_EF;
+const foo = 0b10_00_1111;
+const foo = 0o1_0_44_21;
+const foo = 1_294_28771_2n;
+```
+
+## Pass
+
+```js
+const foo = 1_234_444;
+const foo = 1_234.567_89;
+const foo = 0xAB_CD_EF;
+const foo = 0b1000_1111;
+const foo = 0o10_4421;
+const foo = 1_294_287_712n;
+```
+
+## Options
+
+If you want a custom group size for a specific number type, you can specify it here.
+
+There are 4 options, `hexadecimal`, `binary`, `octal`, `number`, which all refer to their corresponding type. Each of them has to be associated with an object
```

Comment:
Add MDN links for  `hexadecimal` https://github.com/sindresorhus/eslint-plugin-unicorn/blob/master/docs/rules/number-literal-case.md

Reply:
I added links for all types, as they did in the documentation you linked.